packages/ckb-sdk-utils/src/serialization/index.ts

```diff
@@ -14,58 +19,106 @@ const getOffsets = (elmLengths: number[]) => {
-  const body = bytesToHex(bytes)
-  return body
+/**
+ * @name serializeStruct
+ * @type Struct is a fixed-size type: all fields in struct is fixed-size,
+ *              and it has a fixed quantity of fields.
+ * @description The size of a struct is the sum of all fields' size.
+ *              Serialize a struct only need to serialize all fields in it.
+ *              No extra cost for struct itself.
+ *              Fields in a struct are stored in the order they are declared.
+ */
+export const serializeStruct = (struct: Map<string, string | Uint8Array>) => {
+  let res = ''
+  struct.forEach(value => {
+    res += serializeArray(value)
+  })
+  return res
 }
 
-export const serializeArg = (arg: CKBComponents.Hash) => serializeHash(arg)
-
-export const serializeHashType = (hashType: CKBComponents.ScriptHashType) => {
-  if (hashType === 'data') return '00'
-  if (hashType === 'type') return '01'
-  throw new TypeError("Hash type must be either of 'data' or 'type'")
+/**
+ * @name serializeFixVec
+ * @type FixVec, a fixed-size vector is a vector whose items have a fixed size
+ * @tutorial There are two steps of serializing a serializeFixVec
+ *           1. Serialize the length as a 32 bit unsigned integer in little-endian
+ *           2. Serialize all items in it
+ */
+export const serializeFixVec = (fixVec: string | (string | Uint8Array)[]) => {
+  if (typeof fixVec !== 'string' && !Array.isArray(fixVec)) {
+    throw new TypeError('The fixed vector to be serialized should be a string or an array of bytes')
+  }
+  const vec = typeof fixVec === 'string' ? [...hexToBytes(fixVec)].map(b => `0x${b.toString(16)}`) : fixVec
+  const serializedArgVec = vec.map(serializeArray)
+  const header = toHexInLittleEndian(serializedArgVec.length)
+  return `${header}${serializedArgVec.join('')}`
 }
 
-export const serializeArgs = (args: string[]) => {
-  if (!args.length) return '04000000'
-  const serializedArgs = args.map(serializeArg)
-  const headerLength = fullLengthSize + offsetSize * serializedArgs.length
-  const fullLength = toHexInLittleEndian(headerLength + serializedArgs.join('').length / 2)
-  const offsets = getOffsets(serializedArgs.map(arg => arg.length / 2)).map(offset => toHexInLittleEndian(offset))
-  return [fullLength, ...offsets, ...serializedArgs].join('')
+/**
+ * @name serialzieDynVec
+ * @type DynVec, a dynamic-size vector is a vector whose items have a dynamic size
+ * @tutorial There are three steps of serializing a dynvec
+ *           1. Serialize the full size in bytes as a 32 bit unsigned integer in little-endian
+ *           2. Serialize all offset of the items as 32 bit unsigned integer in little-endian
+ *           3. Serialize all items in it
+ */
+export const serializeDynVec = (dynVec: (string | Uint8Array)[]) => {
+  if (!Array.isArray(dynVec)) {
+    throw new TypeError('The dynamic vector to be serialized should be an array of bytes')
+  }
+  const serializedArgVec = dynVec.map(arg => serializeArray(arg))
```

Comment:
like `bodyElms.map(arg => arg.length / 2)` may be more clear

Reply:
I've renamed `serializedArg` to `serializedItem` [commit](https://github.com/nervosnetwork/ckb-sdk-js/pull/342/commits/86cac9aa56df902cffbd49b7ca8c0925586f8792) cuz the inner item in a vector is referenced as `Item` in the RFC.